### PR TITLE
Add the blog post for the beta graduation of HPA ContainerResource type metric

### PR DESCRIPTION
**Pull request by sanposhiho**
The container resource metrics is planning to get graduated to beta in v1.27. So, the feature will be available for many people by default. 
 It’s worth sharing this feature and the motivation for people since this feature is very important to configure safe HPAs for Pod with multiple containers.

---

ref: https://github.com/kubernetes/enhancements/issues/1610

### CODE REVIEW

**Review by gjtempleton — CHANGES_REQUESTED** [review]
A couple of more general points:
* I think we should call out the changes made by https://github.com/kubernetes/kubernetes/pull/116045 given the potential for users to be looking at these events.
* Could we introduce more detail on the metrics, including the labels on the metrics?

**Comment by harshitasao** [comment]
Hello, Comms Lead for the 1.27 release here. This feature blog is tracked for release, the deadline for submitting the draft is on **4th of April**- the sooner the better since there's still editing to be done afterwards. Any doubt, the Comms team is here to help.
Thank you!

**Comment by sanposhiho** [comment]
@gjtempleton Thanks for reviews, please retake a look when have time 🙏

**Comment by harshitasao** [comment]
Hello @sanposhiho, we're doing a global reminder about submitting a draft for review for all opted-in feature blogs. If it's at all possible, it is very helpful for the release team to have drafts submitted for review before the hard deadline date, to better plan the release dates and avoid missing out. Thank you!

**Comment by gjtempleton** [comment]
Thanks for addressing the feedback. Looks good to me from a technical standpoint now.

/lgtm

**Comment by sanposhiho** [comment]
/assign @onlydole

For approval

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:10]:
```suggestion
Kubernetes 1.20 introduced [the ContainerResource type metric](/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics) in HPA.
```

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:20]:
```suggestion
(See [the algorithm details](/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details)
to know how the desired replica number is calculated exactly).
```

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:34]:
```suggestion
HPA already has [the Resource type metric](/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-resource-metrics). 
```

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:73]:
```suggestion
Also, we've improved the observability of HPA controller by exposing some metrics from the kube-controller-manager:
```

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:74]: the name doesn't match the description, is it correct/accurate?

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:75]: This seems should be swapped with the first item?

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:96]:
```suggestion
- [The official document of the ContainerResource type metric](/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics)
```

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (sanposhiho) [reply content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md]: Oops, thanks.

**Comment by sanposhiho** [comment]
@tengqm Thanks for the review. I've fixed it based on your suggestions

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:62]:
```suggestion
- A sidecar container is only providing an auxiliary service such as log shipping.
  If the application does not log very frequently or does not produce logs in its hotpath
  then the usage of the log shipper will not grow.
- A sidecar container which provides authentication. Due to heavy caching
  the usage will only increase slightly when the load on the main container increases.
  In the current blended usage calculation approach this usually results in
  the HPA not scaling up the deployment because the blended usage is still low.
- A sidecar may be injected without resources set which prevents scaling
  based on utilization. In the current logic the HPA controller can only scale
  on absolute resource usage of the pod when the resource requests are not set.
```

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (tengqm) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:88]:
```suggestion
so that everyone can check whether the events came from the resource metric or
the container resource metric (See [the related PR](https://github.com/kubernetes/kubernetes/pull/116045)).
```

**Review by sftim — COMMENTED** [review]
/hold
pending assignment of publication date

**Comment by sanposhiho** [comment]
@tengqm I applied fix based on your suggestion 🙏

**Comment by harshitasao** [comment]
Hello, Comms Lead for v1.27 here. The publication order and date for the Feature Blog series has been [finalized](https://github.com/kubernetes/sig-release/discussions/2208) and the [tracking board](https://github.com/orgs/kubernetes/projects/117/views/4?filterQuery=status%3ATracked+blog-status%3A%22Ready+for+Review%22) is updated.

The publication date for this article is 02-05-2023 (May 02).

Thank you!

File: content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md
Comment (harshitasao) [inline content/en/blog/_posts/2023-04-13-hpa-container-resource-metric.md:4]:
```suggestion
date: 2023-05-02
```
Please also change the file path to match the new date.

**Comment by harshitasao** [comment]
@sanposhiho, friendly reminder for the above fixes.

**Comment by sanposhiho** [comment]
Ah, sorry for unnotice. I'll give a quick fix.

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (nate-double-u) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:57]: spelling:
```suggestion
The resource utilization of each container may not have a direct correlation or may grow at different rates as the load changes.
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (nate-double-u) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:92]: word choice:
```suggestion
you can get deeper insight into which part has a problem, where takes time, how many scaling tends to happen at which time on your cluster etc.
```

**Comment by nate-double-u** [comment]
A couple of things could be improved with spelling and word choice. Still, given its publication date is today, and we are already [pushing other blog posts out](https://github.com/kubernetes/website/pull/40194), I'd like to approve this -- especially since it has a technical LGTM.

Reapplying @gjtempleton's tech lgtm:
/lgtm

Approving from a tech docs lens:
/approve

I'll open a PR to update the spelling and grammar shortly after approval.

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:3]:
```suggestion
title: "Kubernetes 1.27: HorizontalPodAutoscaler ContainerResource type metric moves to beta"
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:8]:
```suggestion
**Author:** [Kensei Nakada](https://github.com/sanposhiho) (Mercari)
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:10]:
```suggestion
Kubernetes 1.20 introduced the [`ContainerResource` type metric](/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics)
in HorizontalPodAutoscaler (HPA).
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:35]:
```suggestion
HPA already had a [Resource type metric](/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-resource-metrics).
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:37]:
```suggestion
You can define the target resource utilization like the following,
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:78]:
```suggestion
For Kubernetes v1.27, the ContainerResource type metric is available by default as described at the beginning
of this article.
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:89]:
```suggestion
which corresponds to `.spec.metrics[*].type` for a HorizontalPodAutoscaler.
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:100]:
```suggestion
This feature is managed by [SIG Autoscaling](https://github.com/kubernetes/community/tree/master/sig-autoscaling). 
```

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:94]:
```suggestion
Another minor stuff, we've changed the `SuccessfulRescale` event's messages
```

**Review by sftim — CHANGES_REQUESTED** [review]
A few, important fixes.

File: content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md
Comment (sftim) [inline content/en/blog/_posts/2023-05-02-hpa-container-resource-metric.md:4]:
```suggestion
date: 2023-05-02T12:00:00+0800
```

**Review by sftim — COMMENTED** [review]
It's late, but perhaps still better to ship than not.

/lgtm
/hold cancel

**Pull request merged**
